### PR TITLE
Update to use new page visit format

### PR DESCRIPTION
**Pull request by tuchfarber**
New page visit format returns 1 page per journal. This prevents us forwarding to a page from a different journal. 

Relies on edx/journals#150

### CODE REVIEW

**Comment by bill-filler** [comment]
Seeing a bunch of problems:
- This doesn't seem to prevent manually going to a page from another journal i.e. http://localhost:1991/4/pages/14 where page 14 doesn't belong to journal 4
- getting into state where clicking journal from index page is taking me to wrong page
Maybe something is messed up in my setup but seems pretty busted

**Comment by bill-filler** [comment]
Is there another PR to prevent loading of another journal's pages? We can't rely only on the last visited page being for the correct journal

File: src/components/JournalPage/index.jsx [hunk 1]
```diff
@@ -22,9 +22,11 @@ import './JournalPage.scss';
 
 
 class JournalPage extends React.Component {
-  componentDidMount() {
+  constructor(props) {
+    super(props);
```
Comment (bill-filler) [inline src/components/JournalPage/index.jsx]: why was this change needed?

**Comment by bill-filler** [comment]
This is working better. It doesn't handle the case where user navigates within a journal, then goes back to About or Index page and re-enters journal as SiteInfo.visitedPages is stale. 
We should update SiteInfo.visitedPages when navigating pages so it will be correct without having to re-fetch SiteInfo from the server.

File: src/components/JournalPage/index.jsx (hunk 1 above)
Comment (tuchfarber) [reply src/components/JournalPage/index.jsx]: The react lifecycle goes:
1) constructor()
2) static getDerivedStateFromProps()
3) render()
4) componentDidMount()

So we were actually requesting the page after we were asking the page to render. This just makes us wait less for the page to render correctly.

**Comment by tuchfarber** [comment]
Last commit addresses locally updating visitedPages.
